perceval/backends/github.py

```diff
@@ -0,0 +1,250 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Alvaro del Castillo San Felix <acs@bitergia.com>
+#
+
+
+
+import json
+import logging
+import os.path
+
+import requests
+
+from ..backend import Backend, BackendCommand
+from ..cache import Cache
+from ..errors import CacheError
+from ..utils import DEFAULT_DATETIME, str_to_datetime
+
+logger = logging.getLogger(__name__)
+
+class GitHub(Backend):
+    """GitHub backend for Perceval
+
+    This class allows the fetch the issues stored in GitHub
+    repository.
+    """
+
+    def __init__(self, owner=None, repository=None, token=None,
+                 cache=None):
+        """ Init a GitHub instance
```

Comment:
El mar, 22-12-2015 a las 17:02 -0800, Santiago Dueñas escribió:
> In perceval/backends/github.py:
> > +from ..cache import Cache
> > +from ..errors import CacheError
> > +from ..utils import DEFAULT_DATETIME, str_to_datetime
> > +
> > +logger = logging.getLogger(__name__)
> > +
> > +class GitHub(Backend):
> > +    """GitHub backend for Perceval
> > +
> > +    This class allows the fetch the issues stored in GitHub
> > +    repository.
> > +    """
> > +
> > +    def __init__(self, owner=None, repository=None, token=None,
> > +                 cache=None):
> > +        """ Init a GitHub instance
> It's prettier and a common practice, too.

Great! So then, I will change it as your proposed (here and in gerrit
backend).

> —
> Reply to this email directly or view it on GitHub.

-- 
Alvaro del Castillo San Félix
acs@bitergia.com - Chief Technical Officer (CTO)
http://www.bitergia.com
"Software metrics for your peace of mind"